src/kernel/thread/stack_protect.c

```diff
@@ -0,0 +1,49 @@
+#include <kernel/thread/stack_protect.h>
+
+#include <assert.h>
+#include <util/binalign.h>
+#include <hal/mmu.h>
+#include <mem/vmem/vmem_alloc.h>
+
+static int enabled = 0;
```

Comment:
I just wanted to make protection configurable in runtime like vmem for example. Do you think it's unnecessary option?